### DISCUSSION TITLE
Error when using mongodb, prisma and google provider with nextauth

### DISCUSSION THREAD

**Discussion by ghost** [prisma/prisma#16975] [Q&A]
I get a very big error when i sign in via google provider, connected with mongodb and prisma...

schema.prisma file...

![Screenshot 2022-12-22 10 17 38 AM](https://user-images.githubusercontent.com/114661530/209065058-e3356cf3-8f01-4675-a55e-d77f8d70692e.png)

Error in the terminal: 

[next-auth][warn][NO_SECRET] 
https://next-auth.js.org/warnings#no_secret
[next-auth][error][adapter_error_getUserByAccount] 
https://next-auth.js.org/errors#adapter_error_getuserbyaccount 
Invalid `p.account.findUnique()` invocation in
/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique(
Error parsing connection string: Database must be defined in the connection string {
  message: '\n' +
    'Invalid `p.account.findUnique()` invocation in\n' +
    '/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45\n' +
    '\n' +
    '   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),\n' +
    '   9 async getUserByAccount(provider_providerAccountId) {\n' +
    '  10     var _a;\n' +
    '→ 11     const account = await p.account.findUnique(\n' +
    'Error parsing connection string: Database must be defined in the connection string',
  stack: 'Error: \n' +
    'Invalid `p.account.findUnique()` invocation in\n' +
    '/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45\n' +
    '\n' +
    '   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),\n' +
    '   9 async getUserByAccount(provider_providerAccountId) {\n' +
    '  10     var _a;\n' +
    '→ 11     const account = await p.account.findUnique(\n' +
    'Error parsing connection string: Database must be defined in the connection string\n' +
    '    at RequestHandler.handleRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35038:13)\n' +
    '    at RequestHandler.handleAndLogRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:34996:12)\n' +
    '    at /workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35542:25\n' +
    '    at async PrismaClient._executeRequest (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36111:22)\n' +
    '    at async PrismaClient._request (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36082:16)\n' +
    '    at async getUserByAccount (/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:29)',
  name: 'Error'
}
[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR] 
https://next-auth.js.org/errors#oauth_callback_handler_error 
Invalid `p.account.findUnique()` invocation in
/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique(
Error parsing connection string: Database must be defined in the connection string Error: 
Invalid `p.account.findUnique()` invocation in
/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique(
Error parsing connection string: Database must be defined in the connection string
    at RequestHandler.handleRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35038:13)
    at RequestHandler.handleAndLogRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:34996:12)
    at /workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35542:25
    at async PrismaClient._executeRequest (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36111:22)
    at async PrismaClient._request (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36082:16)
    at async getUserByAccount (/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:29) {
  name: 'GetUserByAccountError',
  code: undefined
}
[next-auth][error][adapter_error_getUserByAccount] 
https://next-auth.js.org/errors#adapter_error_getuserbyaccount 
Invalid `p.account.findUnique()` invocation in
/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique(
Error parsing connection string: Database must be defined in the connection string {
  message: '\n' +
    'Invalid `p.account.findUnique()` invocation in\n' +
    '/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45\n' +
    '\n' +
    '   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),\n' +
    '   9 async getUserByAccount(provider_providerAccountId) {\n' +
    '  10     var _a;\n' +
    '→ 11     const account = await p.account.findUnique(\n' +
    'Error parsing connection string: Database must be defined in the connection string',
  stack: 'Error: \n' +
    'Invalid `p.account.findUnique()` invocation in\n' +
    '/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45\n' +
    '\n' +
    '   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),\n' +
    '   9 async getUserByAccount(provider_providerAccountId) {\n' +
    '  10     var _a;\n' +
    '→ 11     const account = await p.account.findUnique(\n' +
    'Error parsing connection string: Database must be defined in the connection string\n' +
    '    at RequestHandler.handleRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35038:13)\n' +
    '    at RequestHandler.handleAndLogRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:34996:12)\n' +
    '    at /workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35542:25\n' +
    '    at async PrismaClient._executeRequest (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36111:22)\n' +
    '    at async PrismaClient._request (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36082:16)\n' +
    '    at async getUserByAccount (/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:29)',
  name: 'Error'
}
[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR] 
https://next-auth.js.org/errors#oauth_callback_handler_error 
Invalid `p.account.findUnique()` invocation in
/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique(
Error parsing connection string: Database must be defined in the connection string Error: 
Invalid `p.account.findUnique()` invocation in
/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:45

   8 getUserByEmail: (email) => p.user.findUnique({ where: { email } }),
   9 async getUserByAccount(provider_providerAccountId) {
  10     var _a;
→ 11     const account = await p.account.findUnique(
Error parsing connection string: Database must be defined in the connection string
    at RequestHandler.handleRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35038:13)
    at RequestHandler.handleAndLogRequestError (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:34996:12)
    at /workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:35542:25
    at async PrismaClient._executeRequest (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36111:22)
    at async PrismaClient._request (/workspaces/chat-app/frontend/node_modules/@prisma/client/runtime/index.js:36082:16)
    at async getUserByAccount (/workspaces/chat-app/frontend/node_modules/@next-auth/prisma-adapter/dist/index.js:11:29) {
  name: 'GetUserByAccountError',
  code: undefined

**Comment by nurul3101** [comment]
Hello @fwJayy 👋 

Can you share your connection string format - after obfuscating your username and password from the connection string.
The error message indicates something wrong with the format of connection string.
Are you using MongoDB Atlas to host your database?

**Reply by ghost** [reply]
I am using MongoDB atlas

**Reply by ghost** [reply]
I am kind of new to all this, can you you tell where the connection string format is....
Are you asking for this: mongodb+srv://fwJayy:*********@incog.pao91od.mongodb.net/?retryWrites=true&w=majority

**Reply by nurul3101** [reply]
Can you try this format:
```
mongodb+srv://fwJayy:*********@incog.pao91od.mongodb.net/databasename?retryWrites=true&w=majority
```

After the url and before query parameters try adding the name of your database first.

**Reply by ghost** [reply]
ok one sec

**Reply by ghost** [reply]
> Can you try this format:
> 
> ```
> mongodb+srv://fwJayy:*********@incog.pao91od.mongodb.net/databasename?retryWrites=true&w=majority
> ```
> 
> After the url and before query parameters try adding the name of your database first.

No, the same error

**Reply by nurul3101** [reply]
Strange, what happens if you remove the parameters? Just like this:
```
mongodb+srv://fwJayy:*********@incog.pao91od.mongodb.net/databasename
```

**Reply by ghost** [reply]
Is there anything to do with the schema.prisma file

**Reply by nurul3101** [reply]
In the schema name you have defined env variable as "MONGODB_URI", but in your env file it seems you don't have that variable defined and instead are using "DATABASE_URL", can you replace DATABASE_URL with MONGODB_URI in env file?

**Reply by ghost** [reply]
i changed it right now both the places i  have MONGODB_URI

**Reply by nurul3101** [reply]
Does the initial error go away?

**Reply by ghost** [reply]
yea like the error in the google console when you inspect, that one went....

**Reply by ghost** [reply]
the on in the vscode terminal is still there...

**Comment by ghost** [comment]
![Screenshot 2022-12-22 1 04 13 PM](https://user-images.githubusercontent.com/114661530/209081500-a399496f-2b56-4d65-961e-a890046a5060.png)

**Reply by nurul3101** [reply]
I would recommend you to delete this, as it exposes your GOOGLE_CLIENT_ID and GOOGLE_CLIENT_SECRET.

**Reply by ghost** [reply]
ok one sec

**Reply by ghost** [reply]
should i delete all extentions?

**Reply by nurul3101** [reply]
You can disable them first to identify which extension is causing the error

**Reply by ghost** [reply]
I did everything, still i get the same error in the terminal in vscode... Just the that the error disappears in the google console...

**Reply by nurul3101** [reply]
Did you try restarting the terminal after changing env variable name?

**Reply by ghost** [reply]
yes yes

**Reply by ghost** [reply]
I always generate new prisma table and also run "npm run dev" everytime i do a change...

**Reply by ghost** [reply]
You want the error pic again now...?

**Reply by nurul3101** [reply]
Are you able to connect to your database through a Database IDE like MongoDB Compass?

**Reply by ghost** [reply]
yea i am connected in the compass app

**Comment by ghost** [comment]
![Screenshot 2022-12-22 1 15 12 PM](https://user-images.githubusercontent.com/114661530/209083386-cb3ede98-3174-4475-b5f7-85210a3d1724.png)

**Reply by ghost** [reply]
this is the error in the google console...

**Reply by nurul3101** [reply]
It looks like the error message is coming from one of your extensions, it's not related to your code.

**Reply by ghost** [reply]
oh...
